### PR TITLE
Api integration for homepage and products listing

### PR DESCRIPTION
**Pull request by abhisheksahani-nova**
Code change :- 

1. Added products using api calls at homepage and products listing page.
2. Created context for products data to use across homepage and productlisting page.
3. Add filter feature with custom hook using useReducer

Note :- To navigate to productListing page , use  /products route , Link has not been added , I will do that in next PR .

### CODE REVIEW

**Review by samyakshah3008 — APPROVED** [review]
Clean code with proper semantic tags and good naming conventions 

Suggestion: You can add some comments where necessary later on :) 

Rest is all good ✅

**Comment by samyakshah3008** [comment]
You have some merge conflicts, make sure to fix it out

**Comment by abhisheksahani-nova** [comment]
> Clean code with proper semantic tags and good naming conventions
> 
> Suggestion: You can add some comments where necessary later on :)
> 
> Rest is all good ✅

Thanks for the  review , I will definetely add comments all over the app

File: src/reducer/filterReducer.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    filterProducts = filterProducts.filter(
+      ({ rating }) => rating >= state.fourRatingPlus
+    );
+  }
+
+  return filterProducts;
+}
+
+const useFilter = () => {
+  const [state, dispatch] = useReducer(filterReducer, initialFilterState);
```
Comment (Maianki) [inline src/reducer/filterReducer.js:154]: nitpicking: you can give more meaningful names to these like filters and filtersDispatch.

**Review by Maianki — APPROVED** [review]
Your code looks good. It is readable and modular.

File: src/reducer/filterReducer.js (hunk 1 above)
Comment (abhisheksahani-nova) [reply src/reducer/filterReducer.js:154]: Sure ,I will do that, thanks for the review

**Pull request merged**